src/components/slider/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import {
+  Slider as ChakraSlider,
+  SliderMark as ChakraSliderMark,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+} from '@chakra-ui/react';
+
+type SliderMark<T> = {
+  stepValue?: number;
+  label: string;
+  value: T;
+};
+
+type SliderProps<T> = {
+  useIndentation?: boolean;
+  marks: SliderMark<T>[];
+  defaultValue: SliderMark<T>;
+  onValueChanged: (arg: T) => void;
+};
+
+const getDefaultStepValue = <T,>(
+  marks: SliderMark<T>[],
+  step: number = 1,
+  useIndentation: boolean = false,
+  defaultValue: SliderMark<T>,
+) => {
+  return (
+    (defaultValue && (marks.indexOf(defaultValue) + +useIndentation) * step) ||
```

Comment:
Can we simplify this? One thing that comes to mind is doing an early return in case we have a default value. Also, expressions like `+!useIndentation` are a bit hard to read so maybe be we can be more verbose in favour of improved readability.